src/usb_moded-config.c

```diff
@@ -1061,6 +1061,16 @@ char * config_get_android_manufacturer(void)
     {
         return ssu_name;
     }
+#else
+    /*Trying use /etc/hw-release to get MANUFACTURER_ID*/
+    GKeyFile *hwReleaseFile;
+    GError *error = NULL;
+
+    // Create a new GKeyFile object and a bitwise list of flags.
+    hwReleaseFile = g_key_file_new();
+    if (g_key_file_load_from_file(hwReleaseFile, "/etc/hw-release", G_KEY_FILE_NONE, &error)) {
+        return g_key_file_get_string(hwReleaseFile, NULL, "MANUFACTURER", NULL);
```

Comment:
Does this actually work?

Documentation does not explicitly allow passing NULL group_name.
Code has: g_return_val_if_fail (group_name != NULL, NULL);
Which would indicate that passing null group name is considered an programming error -> some diagnostic logging and NULL return value.

Also, to still allow fallback to config_get_conf_string() below, this should probably return from here only if non-null result was obtained from key file query?